src/util/WalletBackendAdapter.ts

```diff
@@ -188,15 +205,79 @@ export const makeWalletBridgeFromFollowers = (
     marshaller,
   );
 
+  const { notifier: beansOwingNotifier, updater: beansOwingUpdater } =
+    makeNotifierKit<Number | null>(null);
+
   const watchBeansOwing = async () => {
     for await (const { value } of iterateLatest(beansOwingFollower)) {
+      if (isHalted) return;
       beansOwingUpdater.updateState(Number(value));
     }
   };
 
+  // Reads purses from the cosmos bank module. These are not necessarily real
+  // "purses" in the smart wallet contract, because those are lazily
+  // instantiated, but we still need to show the balances.
+  const watchChainBalances = () => {
+    let vbankAssets;
+    let bank;
+
+    const possiblyUpdateBankPurses = () => {
+      if (!vbankAssets || !bank) return;
+
+      const bankMap = new Map<string, string>(
+        bank.map(({ denom, amount }) => [denom, amount]),
+      );
+
+      vbankAssets.forEach(([denom, info]) => {
+        const amount = bankMap.get(denom) ?? 0n;
+        const purseInfo: PurseInfo = {
+          brand: info.brand,
+          currentAmount: AmountMath.make(info.brand, BigInt(amount)),
+          brandPetname: info.issuerName,
+          pursePetname: info.issuerName,
+          displayInfo: info.displayInfo,
+        };
+        brandToPurse.set(info.brand, purseInfo);
+      });
+
+      isBankLoaded = true;
+      updatePurses();
+    };
+
+    const watchBank = async () => {
+      if (isHalted) return;
+      const balances = await queryBankBalances(keplrConnection.address, rpc);
+      bank = balances;
+      possiblyUpdateBankPurses();
+      setTimeout(watchBank, POLL_INTERVAL_MS);
+    };
+
+    const watchVbankAssets = async () => {
+      for await (const { value } of iterateLatest(vbankAssetsFollower)) {
+        if (isHalted) return;
+        vbankAssets = value;
+        possiblyUpdateBankPurses();
+      }
+    };
+
+    void watchVbankAssets();
+    void watchBank();
+  };
+
+  const fetchAgoricBrands = async () => {
+    for await (const { value } of iterateLatest(agoricBrandsFollower)) {
+      if (value) {
```

Comment:
when would the iterator return an object without a `value`?

Reply:
I guess I was afraid it might have an initial value of null or something, but that's more a concern for notifiers, and it would throw anyway in that case. Just removed the conditional